elastic/eui
# [EuiProvider][EuiThemeProvider] `highContrastMode` set up and documentation

## Summary

> [!note]
> This PR merges into a feature branch. As this PR contains some heavy docs changes, I recommend [following along by commit](https://github.com/elastic/eui/pull/8142/files).

This PR sets up the initial architecture for a `highContrastMode` prop/return in `<EuiThemeProvider>` and `useEuiTheme()`. By default, this setting will inherit from the user's system settings with one of 3 possible options:

- `"forced"` (Windows high contrast mode and some browser forced colors extensions)
- `"preferred"` (MacOS high contrast mode)
- `false` (high contrast mode not on)

The provider prop takes a simple true/false (aka `preferred` or `false` - `forced` is something that can only be set externally by the user/system and not by the consumer/application), whereas the internal state returned by our hooks returns more the granular enum for help with styling and other considerations.

**What this PR does not do, and will be handled in future PRs:**

- Address components with shadows that should receive a full border around them instead just a bottom border
- Do anything with forced contrast mode styles
- Perform component-specific audits on custom colors or box shadows that need to be tweaked

## QA

- Go to https://eui.elastic.co/pr_8142/#/theming/high-contrast-mode
- [x] Confirm that while the rest of the page displays normally (shadows, etc.) the first example's table's borders are fully black (or white if in dark mode)
- [MacOS] Go to Display > Increase contrast and check the toggle
  <img width="488" alt="" src="https://github.com/user-attachments/assets/084e4aee-7d15-44e6-9b42-e5537cba35dc">
- [x] Confirm that all shadows on the page disappear, and that all borders become fully black (or white if in dark mode)
- [x] Scroll down to the last example on the page and confirm the border around the text is 2px thick
- Click the theme toggle in the top right corner of the sticky header
- [x] Manually toggle high contrast mode off/on, and confirm it overrides MacOS's system settings

### General checklist

- Browser QA
      - (_[emulate forced colors](https://devtoolstips.org/tips/en/emulate-forced-colors/) if you do not have access to a Windows machine_.)
    ~- [ ] Checked in **mobile**~
    ~- [ ] Checked for **accessibility** including keyboard-only and screenreader modes~
- Docs site QA - N/A
    ~- [ ] Checked **[Code Sandbox](https://codesandbox.io/)** works for any docs examples~
- Code quality checklist
- Release checklist
    ~- [ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)~
- Designer checklist - N/A
